app/src/main/java/org/oppia/android/app/topic/TopicViewModel.kt

```diff
@@ -33,6 +36,38 @@ class TopicViewModel @Inject constructor(
     ).toLiveData()
   }
 
+  private val topicListResultLiveData: LiveData<AsyncResult<PromotedActivityList>> by lazy {
+    topicListController.getPromotedActivityList(
+      ProfileId.newBuilder().setInternalId(internalProfileId).build()
+    ).toLiveData()
+  }
+
+  val numberOfChaptersCompletedLiveData: LiveData<Int> by lazy {
+    Transformations.map(topicListResultLiveData, ::computeNumberOfChaptersCompleted)
+  }
+
+  private fun computeNumberOfChaptersCompleted(
+    topicListResult: AsyncResult<PromotedActivityList>
+  ): Int? {
+    var numberOfChaptersCompleted = 0
+    return when (topicListResult) {
+      is AsyncResult.Failure -> null
+      is AsyncResult.Pending -> null
+      is AsyncResult.Success -> {
+        topicListResult.value.promotedStoryList.recentlyPlayedStoryList.forEach {
+          numberOfChaptersCompleted += it.completedChapterCount
+        }
+        topicListResult.value.promotedStoryList.suggestedStoryList.forEach {
+          numberOfChaptersCompleted += it.completedChapterCount
+        }
+        topicListResult.value.promotedStoryList.olderPlayedStoryList.forEach {
+          numberOfChaptersCompleted += it.completedChapterCount
+        }
+        numberOfChaptersCompleted
+      }
```

Comment:
```suggestion
    return when (topicListResult) {
      is AsyncResult.Failure -> null
      is AsyncResult.Pending -> null
      is AsyncResult.Success -> {
        (topicListResult.value.promotedStoryList.recentlyPlayedStoryList.map {
          it.completedChapterCount
        } + topicListResult.value.promotedStoryList.suggestedStoryList.map {
          it.completedChapterCount
        }  + topicListResult.value.promotedStoryList.olderPlayedStoryList.map {
          it.completedChapterCount
        }).sum()
      }
```

Bit more Kotlin-idiomatic.

Reply:
i believe this function is creating a list of integers, not summing the chapter count

Reply:
We can add parentheses to fix that--I've updated the comment above to clarify.

That being said, I'm not too picky here.